jdaviz/core/template_mixin.py

```diff
@@ -1057,6 +1058,35 @@ def __init__(self, *args, **kwargs):
                                            allowed_type='spatial')
 
 
+class DatasetSpectralSubsetValidMixin(VuetifyTemplate, HubListener):
+    """
+    Adds a traitlet tracking whether self.dataset and self.spectral_subset
+    overlap in the spectral axis.
+
+    Note that if using in another method that is also observing dataset_selected
+    or spectral_subset_selected, that that method could be called before the traitlet
+    is updated.  In that case, call self._check_dataset_spectral_subset_valid() itself
+    directly.  The traitlet can still be used for any warning text in the plugin UI.
+    """
+    spectral_subset_valid = Bool(True).tag(sync=True)
+
+    @observe("dataset_selected", "spectral_subset_selected")
+    def _check_dataset_spectral_subset_valid(self, event={}, return_ranges=False):
+        if self.spectral_subset_selected == "Entire Spectrum":
```

Comment:
```suggestion
        if self.spectral_subset_selected == self.spectral_subset.default_text:
```

Reply:
after offline discussion, we'll reconsider this app-wide as a follow-up effort and then try to be consistent after.